cfn-resources/ldap-configuration/cmd/resource/resource.go

```diff
@@ -0,0 +1,295 @@
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		_, _ = log.Debugf("Update - error: %+v", err)
+		return progressEvents.GetFailedEventByCode(err.Error(), cloudformation.HandlerErrorCodeInvalidRequest), nil
+	}
+
+	// Validate if resource exists
+	_, errPe := Get(client, *currentModel.GroupId)
+	if errPe != nil {
+		return *errPe, nil
+	}
+
+	ldapReq := currentModel.GetAtlasModel()
+
+	LDAPConfigResponse, res, err := client.LDAPConfigurations.Save(context.Background(), *currentModel.GroupId, ldapReq)
+	if err != nil {
+		_, _ = log.Debugf("Create - error: %+v", err)
+		return progressEvents.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+
+	currentModel.CompleteByResponse(*LDAPConfigResponse)
+
+	if err != nil {
+		_, _ = log.Debugf("Update - error: %+v", err)
+		return progressEvents.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+
+	// Response
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		ResourceModel:   currentModel,
+	}, nil
+}
+
+func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	// Validation
+	modelValidation := validator.ValidateModel(DeleteRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		_, _ = log.Debugf("Delete - error: %+v", err)
+		return progressEvents.GetFailedEventByCode(err.Error(), cloudformation.HandlerErrorCodeInvalidRequest), nil
+	}
+
+	// Validate if resource exists
+	_, errPe := Get(client, *currentModel.GroupId)
+	if errPe != nil {
+		return *errPe, nil
+	}
+
+	return executeManualDelete(currentModel), nil
+}
+
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	return handler.ProgressEvent{}, errors.New("not implemented: List")
+}
+
+func executeManualDelete(currentModel *Model) handler.ProgressEvent {
+	log2.Printf("Executing manual save")
+	URL := fmt.Sprintf("https://cloud.mongodb.com/api/atlas/v1.0/groups/%s/userSecurity", *currentModel.GroupId)
+
+	// create a new digest authentication request
+	dr := dac.NewRequest(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey, "PATCH", URL, `{"ldap": {"authenticationEnabled": false,"authorizationEnabled" : false }}`)
+	dr.Header.Set("Content-Type", "application/json")
+	r, err := dr.Execute()
+	if err != nil {
+		_, _ = log.Debugf(fmt.Sprintf("Error creating request %s", err.Error()))
+		return progressEvents.GetFailedEventByCode(fmt.Sprintf("Error creating request %s", err.Error()), cloudformation.HandlerErrorCodeServiceInternalError)
+	}
```

Comment:
Is this calling the atlas api? if yes, you should add the endpoint to the go SDK or fix it instead of calling the endpoint in this way.

Reply:
yes we descuss that with Zuhair, and for timing issues we decided to do a direct call

Reply:
@Adelmar92 can you create a detailed GitHub issue on this (along with proposed solutions if possible) and post to atlas go-client? this way this issue is documented and APIx team can triage appropriately. @andreaangiolillo our SoW on this engagement is expiring this Friday 12/30 so unlikely we will have time to create PR to fix directly in go-client. however, if we do manage to finish early we can then swing back to close out issue entirely. hope this helps.